File: JavaScript/Behavioral/Command.js
/* Command is a behavioral design pattern that turns a 
request into a stand-alone object that contains all 
information about the request. This transformation lets 
you pass requests as a method arguments, delay or queue 
a request’s execution, and support undoable operations. */

// the creation of the command function which can be called
// from diferent interfaces
function Command(data) {
    console.log(data);
}

class Button {
    save(data) {
        Command(data);
    }
}

class MenuItem {
    save(data) {
        Command(data);
    }
}

class Shortcut {
    save(data) {
        Command(data);
    }
}

let button = new Button();
let menuItem = new MenuItem();
let shortcut = new Shortcut();

button.save('Button');
menuItem.save('MenuItem');
shortcut.save('Shortcut');

File: CSharp/Behavioral/Command.cs
namespace Behavioral;

public class Command
{
    public static void CommandMethod(string data)
    {
        Console.WriteLine(data);
    }
    public class Button
    {
        public void Save(string data)
        {
            CommandMethod(data);
        }
    }
    public class MenuItem
    {
        public void Save(string data)
        {
            CommandMethod(data);
        }
    }
    public class Shortcut
    {
        public void Save(string data)
        {
            CommandMethod(data);
        }
    }
}